packages/joy-utils/src/Transport.ts

```diff
@@ -1,6 +1,9 @@
 export abstract class Transport {
-  protected promise<T>(value: T): Promise<T> {
-    return new Promise<T>((resolve, reject) => {
+  protected async promise<T>(value: T, timeout:number = 1): Promise<T> {
```

Comment:
Why do we need this `timeout`?
By default, this `promise` waits for 1 millisecond.

Reply:
So I need a timeout to simulate the API request taking some time to process in the mock data layer transport.

This, in the current setup, is necessary to prevent a race condition between asynchronous data coming in from the data layer transport and the react components refreshing. To cut a long story short, I found that the transport promises are sometimes resolved faster than the react components can render, so their state hook callback is never updated. Even a 1ms timeout fixes the problem completely.

You're right, though: there shouldn't be a global, default timeout. I'll remove it from the default function argument, as suggested, and move it to the transport mock in a more efficient and intuitive way.

Reply:
I should also mention that the final version of the reducer / controller / view pattern we've been talking about (the thing that you called `MediaView` and I called `Controller`) may fix this problem completely, in which case I'll remove all the timeouts.